testing/e2e/clients/application.go

```diff
@@ -0,0 +1,58 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/Azure/aks-app-routing-operator/pkg/util"
+	"github.com/Azure/aks-app-routing-operator/testing/e2e/logger"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	msgraphsdk "github.com/microsoftgraph/msgraph-sdk-go"
+	graphapplications "github.com/microsoftgraph/msgraph-sdk-go/applications"
+	graphmodels "github.com/microsoftgraph/msgraph-sdk-go/models"
+)
+
+// GetServicePrincipalOptions populates a new ServicePrincipalOptions struct with fresh credentials and application/client/servicePrincipal object ids
+func GetServicePrincipalOptions(ctx context.Context, applicationObjectID string, credName string) (*ServicePrincipalOptions, error) {
```

Comment:
What is ServicePrincipalOptions?

Why is this not just called get ServicePrincipal?

Reply:
options is not a descriptive name. it has a new set of creds, and information about the containing application which isn't part of the service principal either so it has more than just a service principal

Reply:
i guess all of those are relevant to the sp, so we could call it that anyway